### ISSUE TITLE
Intellisense doesn't consider ruleset

### ISSUE THREAD

**Issue by molinch** [dotnet/roslyn-analyzers#2712]
#### Analyzer package
https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers

#### Package Version
v2.9.3

#### Diagnostic ID
Ruleset file is correctly used during build, but Intellisense doesn't consider it.
Even worse, if I turn on full solution analysis then absolutely all rules are considered by Intellisense.

#### Repro steps

1. Create Directory.Build.props
```xml
<Project>

  <PropertyGroup>
    <CodeAnalysisRuleSet>$(MSBuildThisFileDirectory)Custom.ruleset</CodeAnalysisRuleSet>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="Microsoft.CodeAnalysis.FxCopAnalyzers" Version="2.9.3" />
  </ItemGroup>

</Project>
```

2. DisableAll.ruleset
![image](https://user-images.githubusercontent.com/5130190/62034768-e4c59e80-b1ee-11e9-9054-e5c46891445d.png)
Remark: It is used as a child ruleset, without disabling rules by default I have issues even in build.
So that's a workaround.

3. Custom.ruleset
![image](https://user-images.githubusercontent.com/5130190/62034810-f3ac5100-b1ee-11e9-8a86-a27806fc8dbb.png)

#### Expected behavior
Only the rules defined in my ruleset should apply during both build and Intellisense.

#### Actual behavior
Build respects the ruleset while Intellisense doesn't.

**Comment by mavasani** [comment]
I am unable to repro this with a simple project (attached) on the latest VS2019 16.3 bits:

[ClassLibrary23.zip](https://github.com/dotnet/roslyn-analyzers/files/3448964/ClassLibrary23.zip)

The project has same ruleset structure and package reference, and has source code that triggers CA1001, which is disabled in DisableAll.ruleset, but bumped to error in primary ruleset. The error shows up both in intellisense and during build.

**Comment by mavasani** [comment]
Closing the issue. Kindly provide a simple repro project if you have one where it still repros for you.

**Comment by mavasani** [comment]
Note that I also tried moving the CodeAnalysisRuleSet property and PackageReference to a Directory.Build.props instead of having them in a project file, and I am still unable to repro your issue.

**Comment by molinch** [comment]
@mavasani Thanks a lot for looking into this.
I played with your sample on both VS2017 and VS2019, with different cases, and indeed I couldn't reproduce our issue.

In our case we had some projects where warnings were configured as error, and some of them didn't have proper WarningsNotAsErrors set (for the code analysis warnings).
Now we simplified our configuration, on all projects and it works as charm, both from Build/Intellisense.

Sorry for the waste of time on this, and thanks again for the investigation.
